### PR TITLE
fix(ui): Calculate fee rebate from full order matching fee

### LINKED ISSUE #2437: Fix `fee_rebate` calculation on order and confirmation view
The fee rebate is incorrectly calculated using the already rebated fee to calculate the rebate from.

### PR DESCRIPTION
**Pull request by holzeis**
fixes #2437

### CODE REVIEW

File: mobile/lib/features/trade/trade_bottom_sheet_confirmation.dart [hunk 1]
```diff
@@ -177,10 +177,10 @@ class TradeBottomSheetConfirmation extends StatelessWidget {
     TextStyle dataRowStyle = const TextStyle(fontSize: 14);
 
     final orderMatchingFee = tradeValues.fee ?? Amount.zero();
-    final feeRebate = referralStatus != null
-        ? Amount((referralStatus.referralFeeBonus * orderMatchingFee.sats).ceil())
+    final feeBeforeRebate = referralStatus != null && orderMatchingFee.sats > 0
+        ? Amount((orderMatchingFee.sats / (1.0 - referralStatus.referralFeeBonus)).floor())
```
Comment (luckysori) [inline mobile/lib/features/trade/trade_bottom_sheet_confirmation.dart:181]: 🤔 I think this can lead to a division-by-zero error.

Comment (holzeis) [reply mobile/lib/features/trade/trade_bottom_sheet_confirmation.dart:181]: If we have a fee rebate of 100%. I'll add a handling

Comment (luckysori) [reply mobile/lib/features/trade/trade_bottom_sheet_confirmation.dart:181]: Yeah. I only mention it because we talked about this scenario earlier today.

Comment (holzeis) [reply mobile/lib/features/trade/trade_bottom_sheet_confirmation.dart:181]: Hmm. But if we have a fee rebate of 100% the order matching fee will be 0.

**Comment by holzeis** [comment]
@luckysori I fixed the issue if there is a fee rebate of 100%. I had to parameterize the `order_matching_fee` function to not deduct the fee_rebate for the front end so that I can show the rebate on the UI.

This will now also work for a 100% fee deduction.

**Review by luckysori — APPROVED** [review]

**Pull request merged**